Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp

```diff
@@ -134,15 +134,43 @@ static WebFrameMap& webFrameMap()
     return map;
 }
 
-static WebKitFrame* webkitFrameGetOrCreate(WebFrame* webFrame)
+static WebKitFrame* webkitFrameGet(WebFrame* webFrame)
 {
     auto wrapperPtr = webFrameMap().get(webFrame);
     if (wrapperPtr)
         return wrapperPtr->webkitFrame();
 
+    return nullptr;
+}
+
+static CString getDocumentLoaderURL(DocumentLoader* documentLoader)
+{
+    if (!documentLoader)
+        return { };
+
+    if (!documentLoader->unreachableURL().isEmpty())
+        return documentLoader->unreachableURL().string().utf8();
+
+    return documentLoader->url().string().utf8();
+}
+
+static WebKitFrame* webkitFrameGetOrCreate(WebFrame* webFrame)
+{
+    if (auto webKitFrame = webkitFrameGet(webFrame))
+        return webKitFrame;
+
     std::unique_ptr<WebKitFrameWrapper> wrapper = makeUnique<WebKitFrameWrapper>(*webFrame);
-    wrapperPtr = wrapper.get();
+    auto wrapperPtr = wrapper.get();
     webFrameMap().set(webFrame, WTFMove(wrapper));
+
+    auto url = getDocumentLoaderURL(webFrame->coreFrame()->loader().provisionalDocumentLoader());
+
+    if (url.isNull())
+        url = getDocumentLoaderURL(webFrame->coreFrame()->loader().documentLoader());
+
+    if (!url.isNull())
+        webkitFrameSetURI(wrapperPtr->webkitFrame(), url);
+
```

Comment:
This could be moved to the WebKitFrame initialization.

Reply:
Are we sure this is correct? Since we don't know at what point of the load we are, maybe it wouold be better to check first if we have a document loader and if not try with the provisional one.

Reply:
Not sure here but most of the test failed if I put documentLoader as first. It seems that both loader exist in the same time but documentLoader contains empty url, and provisionalDocumentLoader contains expected one(at least in many tests)

Reply:
> This could be moved to the WebKitFrame initialization.

Done